### PR TITLE
LPD-28125 Add support for on demand load of translations in ES modules (first part)

### PR DESCRIPTION
**Pull request by izaera**

### CODE REVIEW

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:test:sf

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:report:32592989

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
Running relevant again to compare one run against the other since there are a lot of "unique failures" that are not unique according to the CSV report.

**Comment by izaera** [comment]
ci:report:33768431

**Comment by izaera** [comment]
There are no unique failures in the last report -> https://test-1-29.liferay.com/userContent/jobs/generate-testray-csv/builds/161/testray-results-33826306.csv

Going go forward.

File: modules/_node-scripts/bundle/esbuild/plugins/getLiferayLanguageGetPlugin.mjs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+					}
+
+					let loader = 'jsx';
+
+					if (args.path.endsWith('.ts')) {
+						loader = 'ts';
+					}
+					else if (args.path.endsWith('.tsx')) {
+						loader = 'tsx';
+					}
```
Comment (dsanz) [inline modules/_node-scripts/bundle/esbuild/plugins/getLiferayLanguageGetPlugin.mjs:64]: Is this handling regular .js files too?

Comment (izaera) [reply modules/_node-scripts/bundle/esbuild/plugins/getLiferayLanguageGetPlugin.mjs:64]: Yes -> https://github.com/liferay-frontend/liferay-portal/pull/4300/files/c5f01fb85bae712ced41a832d3995a65468e0d8f#diff-b17c2f8d0ebc020f5e5eb5c882f74ba6f9351bea39f26d772ec1c38fa4e7f953R30

**Comment by izaera** [comment]
ci:forward:force

**Comment by dsanz** [comment]
LGTM ;)

**Pull request closed without merging**